src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -4822,6 +4822,184 @@ .ForwardHelpCategory Cmdlet
 ";
         }
 
+#if UNIX
+        internal static string GetSudoFunctionText()
+        {
+            return @"
+Set-StrictMode -Version Latest
+
+$script:AmbiguousOperators = @('>', '<', '|', '&')
+$script:IgnoredTokens = @(
+    [System.Management.Automation.PSTokenType]::CommandParameter,
+    [System.Management.Automation.PSTokenType]::CommandArgument,
+    [System.Management.Automation.PSTokenType]::NewLine,
+    [System.Management.Automation.PSTokenType]::Comment,
+    [System.Management.Automation.PSTokenType]::StatementSeparator,
+    [System.Management.Automation.PSTokenType]::LineContinuation
+    )
+
+function Get-SudoArgs
+{
+    param
+    (
+        [System.Collections.Generic.List[string]] $AllArgs
+    )
+    $sb = [System.Text.StringBuilder]::new()
+
+    [string] $arg = $AllArgs[0]
+    while ($arg.StartsWith('-'))
+    {
+        $AllArgs.RemoveAt(0)
+        $null = $sb.Append(' ')
+        $null = $sb.Append($arg)
+
+        if ($AllArgs.Count -gt 0)
+        {
+            $arg = $AllArgs[0]
+        }
+        else
+        {
+            break
+        }
+    }
+    return $sb.ToString()
+}
+
+function Get-ScriptBlock
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [System.Collections.Generic.List[string]] $AllArgs,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string] $SudoCmd
+    )
+    [bool] $usePowerShell = $false
+
+    [string] $sudoArgs = Get-SudoArgs -AllArgs $AllArgs
+
+    <# tokenize the remaining arguments to identify the type of expression; native or powershell
```

Comment:
Based on your current implementation as a function - you **should not** utilize the parser at all - a function is passed values, not PowerShell.

If the intent is to invoke PowerShell as root, then a reasonable syntax would be:

```powershell
sudo { $x + $x }
```

This would be unambiguous - you receive a script block, it's meant to run PowerShell, if you receive anything else, you would simply forward on to the native command `sudo`.

It might be worth some @PowerShell/powershell-committee discussion.